frontend/src/pages/articles/articles-feed.tsx

```diff
@@ -1,12 +1,41 @@
+import {
+  useAppDispatch,
+  useAppSelector,
+  useCallback,
+  useEffect,
+} from '~/libs/hooks/hooks.js';
+import { type UserDetailsResponseDto } from '~/packages/users/users.js';
+import { actions as articlesActions } from '~/slices/articles/articles.js';
+
 import { ArticleCard } from './components/components.js';
-import { MOCKED_ARTICLES, MOCKED_USER } from './libs/constants.js';
-
-const ArticlesFeed: React.FC = () => (
-  <>
-    {MOCKED_ARTICLES.map((article) => (
-      <ArticleCard key={article.id} article={article} user={MOCKED_USER} />
-    ))}
-  </>
-);
+import { MOCKED_REACTIONS, MOCKED_TAGS } from './libs/constants.js';
+
+const ArticlesFeed: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { articles } = useAppSelector((state) => state.articles);
+
+  const handleGetFeed = useCallback(() => {
+    void dispatch(articlesActions.loadAll());
+  }, [dispatch]);
+
+  useEffect(() => {
+    handleGetFeed();
+  }, [handleGetFeed]);
+
+  return (
+    <>
+      {articles?.every((article) => article.userDetails) &&
```

Comment:
```suggestion
      {
```

Reply:
If I remove it then there will be an error when switching between tabs because the shape of articles in both cases is different, so when we switch from the feed tab to the my articles tab, for some moment in the state will be my articles which doesn't have author details

Reply:
So can I make the same shape for them or create to different fields in the state?

Reply:
Firstly you can reset articles in state when component is unmounted,
Secondly - yep, you can provide identical shape for them